src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateArray = [];
+
+  let stateCopy = {
+    ...state,
+  };
+
+  for (const part of actions) {
```

Comment:
```suggestion
  for (const action of actions) {
```
I think action sounds better. 
Also, you can make a destructuring to get all the needed props of action (type, extraData, keysToRemove)